crates/proto/src/lib.rs

```diff
@@ -88,3 +88,48 @@ pub use crate::xfer::retry_dns_handle::RetryDnsHandle;
 #[cfg(feature = "backtrace")]
 pub use error::{ENABLE_BACKTRACE, ExtBacktrace};
 pub use error::{ForwardData, ForwardNSData, ProtoError, ProtoErrorKind};
+
+#[cfg(not(feature = "no_std_rand"))]
+pub(crate) use rand::random;
+
+#[cfg(feature = "no_std_rand")]
+pub(crate) use no_std_rand::random;
+#[cfg(feature = "no_std_rand")]
+pub use no_std_rand::seed_rng;
+
+/// A simple shim that allows us to use a [`StdRng`] in `no_std` environments.
+#[cfg(feature = "no_std_rand")]
+mod no_std_rand {
+
+    use core::cell::RefCell;
+
+    use critical_section::Mutex;
+    use once_cell::sync::Lazy;
+    use rand::distr::{Distribution, StandardUniform};
+    use rand::{Rng, SeedableRng, rngs::StdRng};
+
+    static SEEDED_RNG: Lazy<Mutex<RefCell<StdRng>>> =
+        Lazy::new(|| Mutex::new(RefCell::new(StdRng::seed_from_u64(0x050BAD533D))));
```

Comment:
Do we need both the `Mutex` and the `RefCell` here? It seems like either one should be sufficient.

Reply:
The critical section does not provide interior mutability, see https://docs.rs/critical-section/latest/critical_section/struct.Mutex.html#method.borrow

I'll switch the calls to `borrow_ref_mut` though as per the docs